runtime/lua/vim/diagnostic.lua

```diff
@@ -877,7 +877,8 @@ M.handlers.signs = {
 
     local ns = M.get_namespace(namespace)
     if not ns.user_data.sign_ns then
-      ns.user_data.sign_ns = api.nvim_create_namespace('')
+      ns.user_data.sign_ns =
+        api.nvim_create_namespace(string.format('%s/diagnostic/signs', ns.name))
```

Comment:
We don't have a consistent naming convention for internal namespaces. Throughout the code we use `_` (e.g. `vim_lsp_inlay_hint`), `/` (e.g. `treesitter/dev`), and `.` (e.g. `vim.lsp.clangd.1`). We probably should establish one so that we have some consistency, but we can paint that bikeshed in a different PR.

Reply:
LSP seems to use mostly underscore (with slash for snippets); tree-sitter uses mostly slashes (and dots for dev). API documentation uses dashes.

Of the two, LSP is more stable (and tree-sitter experimental, so breaking changes are not as difficult).

But in general, I feel that this is a case of "follow your own conscience", and I'd like to see guidance in `dev_style.txt` along the lines of
```
1. try to be consistent within a module;
2. if in doubt, prefer <foo>;
3. no need to rewrite old modules to enforce (cross-module) consistency
```